lib/evaluate.js

```diff
@@ -1,20 +1,57 @@
 import {compile, compileSync} from './core.js'
 
 /* eslint-disable no-new-func */
+
+/**
+ * @typedef {import("vfile").VFileCompatible} VFileCompatible
+ * @typedef {import("unified").Pluggable} Pluggable
+ * @typedef {import('./core.js').ProcessorOptions} ProcessorOptions
+ *
+ * @typedef RunnerOptions
+ * @property {*} [Fragment]
+ * @property {*} [jsxs]
+ * @property {*} [recastPlugins]
+ * @property {*} [useMDXComponents]
+ *
+ * @typedef {ProcessorOptions & RunnerOptions} ProcessorAndRunnerOptions
+ * @typedef {{ [name: string]: any }} ComponentMap
+ * @typedef {{ [props: string]: any, components?: ComponentMap }} XDMProps
```

Comment:
applied the first, can you expand what you mean by the second?

Reply:
The result of `evaluate()` at `.default` is a component: a function that takes `MDXContentProps` and returns a call to `jsx`: in React, I’m sure there’s a type for that.

Now, assuming `jsx` had types in React, then there’d be a type for its return value, right? Then I’m guessing that you could take that value and define it as `.default`s return value too?

Illustrated in another way: `jsx` is a function: `f() = x`. So the return value of `.default` is also always `x`

(and for my comment I forgot that `pragma` is not an option for `evaluate`, and the return value of `jsxs` and `jsx` are the same)

Reply:
> The result of evaluate() at .default is a component: a function that takes MDXContentProps

Right, and that is the `ExportMap` type on line 19

> Now, assuming jsx had types in React, then there’d be a type for its return value, right?

For react yes,
one that is consistent across React, Preact, and Vue, I'm not so sure.
I think there are differences (it would take some experimentation to check), and if there are differences, it would be difficult to capture even with conditional types watching what was passed to `pragma`.

Reply:
There would be difference but the result of that `f() = x` should be accessible though?

If someone does `evaluateSync({jsx: f}).default()` the output type is still `x`, and `jsx` is a required option.

I understand that all the things passed as `jsx` will be different, but they’re required to be functions, so that might help?

Reply:
Preact’s runtime does have types: https://github.com/preactjs/preact/blob/9250ac91891d19ec988938971b6767697bcb028e/jsx-runtime/src/index.d.ts#L11-L17